projects/packages/analyzer/src/api/class-data-store.php

```diff
@@ -0,0 +1,94 @@
+<?php
+namespace Automattic\Jetpack\Analyzer;
+
+class Model {
+	private $db_file = 'data-store.json';
+	private $content;
+
+	public function __construct() {
+		$this->db_file = dirname( __DIR__ ) . 'data-store.json';
+	}
+
+	public function get_status() {
+		$this->load();
+		return $this->content;
+	}
+
+	public function toggle_status() {
+		$this->load();
+		if ($this->content['status'] === 'finished') {
+			$this->content['status'] = 'in_progress';
+		} else {
+			$this->content['status'] = 'finished';
+		}
+		$this->persist();
+	}
+
+	public function get() {
+		return $this->content;
+	}
+
+	public function update_process( $increment, $started_plugin = null, $finished_plugin = null ) {
+		$this->load();
+		if ( ! isset( $this->content['progress'] ) ) {
+			$this->content['progress'] = array(
+				'plugins_left' => 0,
+				'in_progress'  => array(),
+			);
+		}
+
+		$current_list = $this->content['progress']['in_progress'];
+
+		if ( null !== $finished_plugin && false !== ( $key = array_search( $finished_plugin, $current_list ) ) ) {
+			array_splice( $current_list, $key, 1 );
+		}
+
+		if ( null !== $started_plugin ) {
+			array_push( $current_list, $started_plugin );
+		}
+
+		$this->content['progress']['in_progress']   = $current_list;
+		$this->content['progress']['plugins_left'] += $increment;
+		$this->persist();
+	}
+
+	public function load_result() {
+		$this->load();
+		$warn_folder = dirname( dirname( __DIR__ ) ) . '/output/warnings';
+		$warnings    = scandir( $warn_folder );
+		unset( $warnings[0] );
+		unset( $warnings[1] );
```

Comment:
Assuming that `.` and `..` are always the first two entries?

Reply:
yeah. definitely an ugly approach 😬 Probably something like `array_diff(scandir( $warn_folder ), array('.', '..'));` would be more explicit.